paddle/fluid/inference/api/analysis_config.cc

```diff
@@ -235,13 +234,13 @@ void AnalysisConfig::Update() {
   }
 
   if (use_mkldnn_) {
+#ifdef PADDLE_WITH_MKLDNN
     if (!enable_ir_optim_) {
       LOG(ERROR)
           << "EnableMKLDNN() only works when IR optimization is enabled.";
+    } else {
+      pass_builder()->EnableMKLDNN();
     }
```

Comment:
Could we remove line 244-247? Since I see `LOG(ERROR) << "Please compile with MKLDNN first to use MKLDNN";use_mkldnn_ = false;` is already in `pass_builder()->EnableMKLDNN();`
Could we remove line 258-261 for the same reason?

Reply:
Right. Done.